### ISSUE TITLE
Use XDG_CONFIG_HOME on MacOS

### ISSUE THREAD

**Issue by cideM** [shibukawa/configdir#9]
I don't have data on this but I am pretty sure that most people place scripts in unix style directories rather than `/Library`. As such, quite a few people probably use `XDG_CONFIG_HOME` even on MacOS. Yet this library doesn't support that, which means every CLI app relying on this library also doesn't (https://github.com/jesseduffield/lazygit/issues/641)

**Comment by wakatara** [comment]
Yeah, I am just looking at this library now and trying to figure out how to override that. I think there should be the option for override for OSX apps. Definitely CLI apps want to be going in the XDG_CONFIG_HOME. 

Course, I was planning on trying to just override this with an `if XDG || Darwin {config.XDG}` type thingy. Still looking into how to easily do that, but yes, I agree with the override (or instructions on how we should be doing it "canonically" with the lib.

**Comment by raine** [comment]
`/Library/Application Support` doesn't appear standard at all for CLI tools.

I don't have a single CLI tool with configuration there.
